nat-lab/tests/telio.py

```diff
@@ -534,6 +534,7 @@ async def run(
     ) -> AsyncIterator["Client"]:
         if isinstance(self._connection, DockerConnection):
             start_tcpdump(self._connection.container_name())
+            await self.clear_core_dumps()
```

Comment:
Actually I'm wondering if it's better to clear core dumps only after collecting them,
not before the test run, in case if something crashed in the previous run and we failed to collect them..

Reply:
It is common to either do both, or just at the start. The idea being, that in case things are crashing -> we cannot be sure that cleanup will be executed reliably.

Of course in this case, the crashing process and the one collecting dumps are two different ones, so the idea above is more rule of thumb rather then something strong.

Reply:
Good point I was thinking along those lines too. I think I will keep it at the beginning for now, In case we see that something is crashing before and we need those logs, we can change it after.

Reply:
Oh, I just realised that, if we do cleanup non-conditionally (whether it is being run in CI or not) it helps to avoid cleanup after test failure when running outside of CI, as it allows for devs to gather the coredumps manually from containers.

Reply:
Good point, I'll add `if (os.environ.get("NATLAB_SAVE_LOGS") is None)` here too